job.c

```diff
@@ -346,7 +347,9 @@ static int handle_line(void *data, const char *line)
 	if (is_http_url(line) || is_cue_url(line)) {
 		add_url(line);
 	} else {
-		char *absolute = path_absolute_cwd(line, data);
+		char *absolute = pl_env_var(line, NULL)
+			? pl_env_restore(line)
+			: path_absolute_cwd(line, data);
```

Comment:
Is it possible to avoid substituting conditional expressions, and just substitute functions which handle both cases internally? The way it is now looks not very clean -- and makes adding new places that load/save filenames a bit harder.

Reply:
Yes, it's possible, but I did choose the current way for a few reasons:

1. This function shouldn't be used arbitrarily, since it is only needed for filenames cmus will read again later. Other file paths should be the resolved ones. If I was to move this into another function, what would it be named to make this clear? 
2. There are currently only two places this will be needed (cache and job), and it's unlikely there will be more. And, each of these requires different path handling (one needs the filename to be made absolute based on the working directory, the other uses it as-is). The way they handle freeing is also different too. Adding a function which combines both adds unnecessary abstraction for little benefit IMO.
3. This keeps `pl_env.{c,h}` purely concerned with doing path manipulation rather than also deciding when it needs to be done.

If you still think it'll be cleaner, I can change it.

Reply:
@flyingmutant?